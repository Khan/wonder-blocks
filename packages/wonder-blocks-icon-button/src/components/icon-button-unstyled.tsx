import * as React from "react";
import {Link, useInRouterContext} from "react-router-dom-v5-compat";

import {addStyle, keys} from "@khanacademy/wonder-blocks-core";
import {isClientSideUrl} from "@khanacademy/wonder-blocks-clickable";
import {StyleSheet} from "aphrodite";
import type {IconButtonProps} from "../util/icon-button.types";

const StyledA = addStyle("a");
const StyledButton = addStyle("button");
const StyledLink = addStyle(Link);

type Props = Omit<IconButtonProps, "icon"> & {
    /**
     * The button content.
     */
    children: React.ReactNode;
    /**
     * URL to navigate to.
     *
     * Used to determine whether to render an `<a>` or `<button>` tag. Also
     * passed in as the `<a>` tag's `href` if present.
     */
    href?: string;
    /**
     * Listens for keydown events on the button. This is useful for preventing
     * default behavior when the user presses the spacebar or enter key.
     */
    onKeyDown?: (e: React.KeyboardEvent) => unknown;
    /**
     * Listens for keyup events on the button. This is useful for triggering
     * actions when the user presses the spacebar or enter key.
     */
    onKeyUp?: (e: React.KeyboardEvent) => unknown;
    /**
     * When the button is in a pressing state. This is useful for keyboard
     * interactions, so we can provide visual feedback to the user.
     */
    onPress?: (isPressing: boolean) => unknown;
};

export const IconButtonUnstyled: React.ForwardRefExoticComponent<
    Props &
        React.RefAttributes<typeof Link | HTMLButtonElement | HTMLAnchorElement>
> = React.forwardRef<
    typeof Link | HTMLButtonElement | HTMLAnchorElement,
    Props
>(function IconButtonUnstyled(props: Props, ref) {
    const {
        children,
        disabled,
        href,
        onPress,
        skipClientNav,
        style,
        testId,
        type = "button",
        ...restProps
    } = props;
    const inRouterContext = useInRouterContext();

    const handleKeyDown = React.useCallback(
        (e: React.KeyboardEvent) => {
            const key = e.key;
            // Prevent default behavior for space and enter keys on
            // buttons. We let the browser handle the default behavior
            // for links, which is to activate the link on `Enter`.
            if (!href && (key === keys.enter || key === keys.space)) {
                e.preventDefault();
                onPress?.(true);
            }
        },
        [href, onPress],
    );

    const handleKeyUp = React.useCallback((e: React.KeyboardEvent) => {
        const key = e.key;
        if (!href && (key === keys.enter || key === keys.space)) {
            if (restProps.onClick) {
                restProps.onClick(e);
                onPress?.(false);
            }
        }
    }, []);

    const commonProps = {
        "data-testid": testId,
        style: [styles.shared, style],
        onKeyDown: handleKeyDown,
        onKeyUp: handleKeyUp,
        ...restProps,
    } as const;

    if (href && !disabled) {
        return inRouterContext && !skipClientNav && isClientSideUrl(href) ? (
            <StyledLink
                {...commonProps}
                to={href}
                ref={ref as React.Ref<typeof Link>}
            >
                {children}
            </StyledLink>
        ) : (
            <StyledA
                {...commonProps}
                href={href}
                ref={ref as React.Ref<HTMLAnchorElement>}
            >
                {children}
            </StyledA>
        );
    } else {
        return (
            <StyledButton
                type={type}
                {...commonProps}
                onClick={disabled ? undefined : restProps.onClick}
                aria-disabled={disabled}
                ref={ref as React.Ref<HTMLButtonElement>}
            >
                {children}
            </StyledButton>
        );
    }
});

const styles = StyleSheet.create({
    shared: {
        position: "relative",
        display: "inline-flex",
        alignItems: "center",
        justifyContent: "center",
        boxSizing: "border-box",
        padding: 0,
        cursor: "pointer",
        border: "none",
        outline: "none",
        textDecoration: "none",
        background: "none",
        margin: 0,
        // This removes the 300ms click delay on mobile browsers by indicating that
        // "double-tap to zoom" shouldn't be used on this element.
        touchAction: "manipulation",
    },
});
