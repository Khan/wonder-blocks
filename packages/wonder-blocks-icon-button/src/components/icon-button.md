Primary, secondary, and tertiary IconButton examples:
```js
import {View} from "@khanacademy/wonder-blocks-core";
import {icons} from "@khanacademy/wonder-blocks-icon";
import IconButton from "@khanacademy/wonder-blocks-icon-button";
import {Strut} from "@khanacademy/wonder-blocks-layout";

<View style={{flexDirection: "row"}}>
    <IconButton
        icon={icons.search}
        aria-label="search"
        onClick={(e) => console.log("hello")}
    />
    <Strut size={16} />
    <IconButton
        icon={icons.search}
        aria-label="search"
        kind="secondary"
        onClick={(e) => console.log("hello")}
    />
    <Strut size={16} />
    <IconButton
        icon={icons.search}
        aria-label="search"
        kind="tertiary"
        href="/search"
    />
    <Strut size={16} />
    <IconButton
        disabled={true}
        icon={icons.search}
        aria-label="search"
        onClick={(e) => console.log("hello")}
    />
    <Strut size={16} />
    <IconButton
        disabled={true}
        icon={icons.search}
        aria-label="search"
        href="/search"
    />
</View>
```

An IconButton on a dark background. Only the primary kind is allowed.
```js
import Color from "@khanacademy/wonder-blocks-color";
import {View} from "@khanacademy/wonder-blocks-core";
import {icons} from "@khanacademy/wonder-blocks-icon";
import IconButton from "@khanacademy/wonder-blocks-icon-button";
import {StyleSheet} from "aphrodite";

const styles = StyleSheet.create({
    row: {
        flexDirection: "row",
        backgroundColor: Color.darkBlue,
        padding: 10,
    },
});

<View style={[styles.row]}>
    <IconButton
        icon={icons.search}
        aria-label="search"
        light={true}
        onClick={(e) => console.log("hello")}
    />
</View>
```
