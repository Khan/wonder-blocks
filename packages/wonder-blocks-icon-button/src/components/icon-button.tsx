import * as React from "react";
import {__RouterContext} from "react-router";

import {getClickableBehavior} from "@khanacademy/wonder-blocks-clickable";
import type {IconAsset, IconSize} from "@khanacademy/wonder-blocks-icon";
import type {AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
import {Link} from "react-router-dom";
import IconButtonCore from "./icon-button-core";

export type SharedProps = Partial<Omit<AriaProps, "aria-disabled">> & {
    /**
     * A Wonder Blocks icon asset, an object specifing paths for one or more of
     * the following sizes: small, medium, large, xlarge.
     */
    icon: IconAsset;
    /**
     * The color of the icon button, either blue or red.
     */
    color?: "default" | "destructive";
    /**
     * The kind of the icon button, either primary, secondary, or tertiary.
     *
     * In default state:
     * - Primary icon buttons are color: props.color
     * - Secondary buttons are offBlack
     * - Tertiary buttons are offBlack64
     *
     * In the hover/focus/press states, all variants have a border.
     */
    kind?: "primary" | "secondary" | "tertiary";
    /**
     * Whether the icon button is on a dark/colored background.
     */
    light?: boolean;
    /**
     * Whether the icon button is disabled.
     */
    disabled?: boolean;
    /**
     * Test ID used for e2e testing.
     */
    testId?: string;
    /**
     * Optional custom styles.
     */
    style?: StyleType;
    // TODO(yejia): use this if ADR #47 has been implemented
    /*
    style?: Style<Exact<{
        width?: number | string
        position: Position,
        ...MarginStyles,
        ...FlexItemStyles,
    }>>,
    */

    /**
     * Adds CSS classes to the IconButton.
     */
    className?: string;
    // NOTE(jeresig): Currently React Docgen (used by Styleguidist) doesn't
    // support ... inside of an exact object type. Thus we had to move the
    // following propers into this SharedProps, even though they should be
    // external. Once that's fixed we can split them back apart.

    /**
     * URL to navigate to.
     *
     * Note: Either href or onClick must be defined
     */
    href?: string;
    // TODO(WB-1262): only allow this prop when `href` is also set.
    /**
     * A target destination window for a link to open in.
     */
    target?: "_blank";
    /**
     * Specifies the type of relationship between the current document and the
     * linked document. Should only be used when `href` is specified. This
     * defaults to "noopener noreferrer" when `target="_blank"`, but can be
     * overridden by setting this prop to something else.
     */
    rel?: string;
    /**
     * Set the tabindex attribute on the rendered element.
     */
    tabIndex?: number;
    /**
     * Whether to avoid using client-side navigation.
     *
     * If the URL passed to href is local to the client-side, e.g.
     * /math/algebra/eval-exprs, then it tries to use react-router-dom's Link
     * component which handles the client-side navigation. You can set
     * `skipClientNav` to true avoid using client-side nav entirely.
     *
     * NOTE: All URLs containing a protocol are considered external, e.g.
     * https://khanacademy.org/math/algebra/eval-exprs will trigger a full
     * page reload.
     */
    skipClientNav?: boolean;
    /**
     * Function to call when button is clicked.
     *
     * This callback should be used for things like marking BigBingo
     * conversions. It should NOT be used to redirect to a different URL or to
     * prevent navigation via e.preventDefault(). The event passed to this
     * handler will have its preventDefault() and stopPropagation() methods
     * stubbed out.
     *
     * Note: onClick is optional if href is present, but must be defined if
     * href is not
     */
    onClick?: (e: React.SyntheticEvent) => unknown;
    /**
     * One of `xsmall` (24px, 16px icon), `small` (32px, 16px icon), `medium` (40px, 24px icon),
     * `large` (60px, 48px icon), or `xlarge` (108px, 96px icon).
     */
    size?: IconSize;
};

/**
 * An IconButton is a button whose contents are an SVG image.
 *
 * To use, supply an onClick function, a wonder-blocks icon asset (see
 * the Icon section) and an aria-label to describe the button functionality.
 * Optionally specify href (URL), clientSideNav, color
 * (Wonder Blocks Blue or Red), kind ("primary", "secondary", or "tertiary"),
 * light (whether the IconButton will be rendered on a dark background),
 * disabled , test ID, and custom styling.
 *
 * The size of an IconButton matches the size of icon it wraps which is 24x24
 * pixels.  The focus ring which is displayed on hover and focus is much
 * larger but does not affect its size.  This matches the behavior of Button.
 *
 * IconButtons require a certain amount of space between them to ensure the
 * focus rings don't overlap.  The minimum amount of spacing is 16px, but
 * you should refer to the mocks provided by design.  Using a Strut in between
 * IconButtons is the preferred way to for adding this spacing.
 *
 * Many layouts require alignment of visual left (or right) side of an
 * IconButton.  This requires a little bit of pixel nudging since each icon
 * as a different amount of internal padding.
 *
 * See the Toolbar documentation for examples of IconButton use that follow
 * the best practices described above.
 *
 * ```js
 * import {icons} from "@khanacademy/wonder-blocks-icon";
 * import IconButton from "@khanacademy/wonder-blocks-icon-button";
 *
 * <IconButton
 *     icon={icons.anIcon}
 *     aria-label="An Icon"
 *     onClick={(e) => console.log("Hello, world!")}
 * />
 * ```
 */
const IconButton: React.ForwardRefExoticComponent<
    SharedProps &
        React.RefAttributes<typeof Link | HTMLButtonElement | HTMLAnchorElement>
> = React.forwardRef<
    typeof Link | HTMLButtonElement | HTMLAnchorElement,
    SharedProps
>((props: SharedProps, ref) => {
    const {
        onClick,
        href,
        skipClientNav,
        tabIndex,
        target,
        color = "default",
        kind = "primary",
        light = false,
        disabled = false,
        size = "medium",
        ...sharedProps
    } = props;
    const renderClickableBehavior = (router: any): React.ReactNode => {
        const ClickableBehavior = getClickableBehavior(
            href,
            skipClientNav,
            router,
        );

        return (
            <ClickableBehavior
                disabled={disabled}
                href={href}
                onClick={onClick}
                role="button"
                target={target}
            >
                {(state, {...childrenProps}) => {
                    return (
                        <IconButtonCore
                            {...sharedProps}
                            {...state}
                            {...childrenProps}
                            color={color}
                            disabled={disabled}
                            kind={kind}
                            href={href}
                            light={light}
                            size={size}
                            skipClientNav={skipClientNav}
                            target={target}
                            tabIndex={tabIndex}
                            ref={ref}
                        />
                    );
                }}
            </ClickableBehavior>
        );
    };

    return (
        <__RouterContext.Consumer>
            {(router) => renderClickableBehavior(router)}
        </__RouterContext.Consumer>
    );
});

export default IconButton;
