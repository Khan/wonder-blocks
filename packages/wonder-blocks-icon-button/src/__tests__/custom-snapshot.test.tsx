import * as React from "react";
import {render} from "@testing-library/react";

import MagnifyingGlass from "@phosphor-icons/core/regular/magnifying-glass.svg";

import IconButtonCore from "../components/icon-button-core";
import {IconButtonSize} from "../components/icon-button";

const defaultHandlers = {
    onClick: () => void 0,
    onMouseEnter: () => void 0,
    onMouseLeave: () => void 0,
    onMouseDown: () => void 0,
    onMouseUp: () => void 0,
    onTouchStart: () => void 0,
    onTouchEnd: () => void 0,
    onTouchCancel: () => void 0,
    onKeyDown: () => void 0,
    onKeyUp: () => void 0,
    onFocus: () => void 0,
    onBlur: () => void 0,
} as const;

describe("IconButtonCore", () => {
    for (const kind of ["primary", "secondary", "tertiary"] as const) {
        for (const color of ["default", "destructive"] as const) {
            for (const size of [
                "xsmall",
                "small",
                "medium",
            ] as IconButtonSize[]) {
                for (const light of kind === "primary"
                    ? [true, false]
                    : [false]) {
                    for (const state of ["disabled"] as const) {
                        const disabled = state === "disabled";
                        const stateProps = {
                            disabled,
                        } as const;
                        test(`kind:${kind} color:${color} size:${size} light:${String(
                            light,
                        )} ${state}`, () => {
                            const {container} = render(
                                <IconButtonCore
                                    icon={MagnifyingGlass}
                                    aria-label="search"
                                    kind={kind}
                                    color={color}
                                    light={light}
                                    size={size}
                                    tabIndex={disabled ? -1 : 0}
                                    {...stateProps}
                                    {...defaultHandlers}
                                />,
                            );

                            expect(container).toMatchSnapshot();
                        });
                    }
                }
            }
        }
    }
});
