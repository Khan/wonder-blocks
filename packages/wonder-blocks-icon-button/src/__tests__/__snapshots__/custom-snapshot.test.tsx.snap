// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonCore kind:primary color:default size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:default size:medium light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:default size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:default size:small light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:default size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:default size:xsmall light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:medium light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:small light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:primary color:destructive size:xsmall light:true disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(255, 255, 255, 0.5); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:default size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:default size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:default size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:destructive size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:destructive size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:secondary color:destructive size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:default size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:default size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:default size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:destructive size:medium light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 40px; width: 40px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:destructive size:small light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 32px; width: 32px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 24px; height: 24px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;

exports[`IconButtonCore kind:tertiary color:destructive size:xsmall light:false disabled 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="search"
    class=""
    style="margin: -8px; position: relative; display: inline-flex; align-items: center; justify-content: center; box-sizing: border-box; padding: 0px; cursor: not-allowed; outline: none; text-decoration: none; background: none; height: 24px; width: 24px; color: rgba(33, 36, 44, 0.32); border-radius: 4px;"
    tabindex="-1"
    type="button"
  >
    <span
      class=""
      style="display: inline-block; vertical-align: text-bottom; flex-shrink: 0; flex-grow: 0; mask-size: 100%; mask-repeat: no-repeat; mask-position: center; background-color: currentColor; width: 16px; height: 16px; mask-image: url(magnifying-glass.svg);"
    />
  </button>
</div>
`;
